### PR TITLE
Oracle collector functional tests

### PR DESCRIPTION
**Pull request by sethsamuel**
### What does this PR do?
Adds functional tests for the Oracel collector. These tests run against a local docker instance for developers, and against a GitLab service on CI.

### Motivation
We want to have more confidence in our collector's reliability by enforcing tests on CI, as well as improving the developer experience when running local tests.

### Describe how to test/QA your changes
To run locally: `inv oracle.test`. This must be done on an x86 machine as the Oracle image runs extremely slowly under emulation.

### CODE REVIEW

File: .gitlab/functional_test/oracle.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+oracle:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:docker"]
+  stage: functional_test
+  needs: ["go_deps"]
+  services:
+    - alias: "oracle"
+      name: "registry.ddbuild.io/images/mirror/oracle:${DBMS_VERSION}"
+      variables:
+        ORACLE_PWD: "datad0g"
```
Comment (amenasria) [inline .gitlab/functional_test/oracle.yml:10]: Double checking for safety: this is clear on purpose, right?

File: tasks/oracle.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+def test(ctx, verbose=False) -> None:
+    """
+    Runs oracle functional tests against a containerized database.
+    """
+
+    if not os.environ.get("CI") and not os.environ.get("SKIP_DOCKER"):
+        start_docker(ctx, verbose)
+
+    try:
+        # ctx.run("docker ps")
```
Comment (amenasria) [inline tasks/oracle.py:18]: **🥜 nitpick:** ‏
```suggestion

```

File: tasks/oracle.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        os.environ["ORACLE_TEST_PORT"] = "1521"
+        if os.environ.get("CI"):
+            os.environ["ORACLE_TEST_SERVER"] = "oracle"
+        else:
+            os.environ["ORACLE_TEST_SERVER"] = "localhost"
+
+        # ctx.run("echo port=$ORACLE_TEST_PORT server=$ORACLE_TEST_SERVER")
```
Comment (amenasria) [inline tasks/oracle.py:25]: **🥜 nitpick:** ‏
```suggestion

```

**Review by amenasria — APPROVED** [review]
A few comments, otherwise LGTM !

**Review by neko-dd — APPROVED** [review]

File: .gitlab/functional_test/oracle.yml (hunk 1 above)
Comment (sethsamuel) [reply .gitlab/functional_test/oracle.yml:10]: Yes, it's just a local testing container and cleartexting it makes the ops a lot easier.

File: pkg/collector/corechecks/oracle/sysmetrics_test.go [hunk 4]
```diff
@@ -14,6 +14,8 @@ import (
 )
 
 func TestSysmetrics(t *testing.T) {
+	t.Skip("sysmetrics can take several minutes to return the correct number")
```
Comment (nenadnoveljic) [inline pkg/collector/corechecks/oracle/sysmetrics_test.go:17]: Could it be that the time is exactly two minutes (two snapshots 60s apart)? If the initialization time is deterministic, we could hard code it.

**Review by nenadnoveljic — APPROVED** [review]

File: pkg/collector/corechecks/oracle/sysmetrics_test.go (hunk 4 above)
Comment (sethsamuel) [reply pkg/collector/corechecks/oracle/sysmetrics_test.go:17]: I had some luck with reducing collection time and calling `Run` multiple times but it feels a bit fragile. Going to leave it for a later improvement.

**Pull request merged**